charts/pravega/templates/clusterrole.yaml

```diff
@@ -2,8 +2,7 @@
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: {{ .Values.serviceAccount.name }}
-  namespace: {{ .Release.Namespace }}
+  name: {{ template "pravega.fullname" . }}-{{ .Release.Namespace }}
```

Comment:
I would suggest having a  pravega-related  suffix here (`-pravega`?), on top of pravega cluster name and namespace.
Reasoning: There might be other components installed to the cluster that follow the pattern of `<custom-resource-name>-<namespace>`. If pravega is one of the components belonging to the same project, there is non-zero probability all custom resources would be named after the project name.

Reply:
`name: {{ template "pravega.fullname" . }}-pravega-{{ .Release.Namespace }}`  like this?

Reply:
yes, like this

Reply:
Made the changes

Reply:
@jkhalack @anishakj I am not sure if adding `pravega` additionally to the name would be required since that is already handled by `{{ template "pravega.fullname" . }}`

Reply:
If I am not mistaken, `pravega.fullname` would contain a `pravega` substring only in the case when the user has not overriden it.

Reply:
@jkhalack My understanding is if `fullnameoverride` is not used, and chart name is not pravega `pravega` is appended. But user can set the name using `fullnameoverride` in that case `pravega` is not appended. Please refer [this ](https://github.com/pravega/charts/blob/master/charts/pravega/templates/_helpers.tpl#L14-#L25)

Reply:
Reverted the changes and  updated documentation

Reply:
I am fine with documenting the possibility of name clashing when using `fullnameoverride`.